In [1]:
name = "Ana"
hourly_wage = 20.0
weekly_hours = 32.0
weekly_wage = hourly_wage * weekly_hours
print(f"{name} earns ${weekly_wage} per week.")

Ana earns $640.0 per week.


In [2]:
temps = [61, 65, 72, 58, 70, 74, 68]
warm = 0
for t in temps:
    if t > 65:
        warm = warm + 1
print(warm)

4


In [3]:
def to_celsius(f):
    return (f - 32) * 5/9

print(round(to_celsius(temps[0]), 1))

16.1


In [4]:
movie = {"title": "Everything Everywhere All at Once", "year": 2022, "rating": 8.8}
for key, value in movie.items():
    print(key, ":", value)

title : Everything Everywhere All at Once
year : 2022
rating : 8.8


In [5]:
import pandas as pd
df = pd.read_csv("chickwts(1).csv", index_col=0)
df.head()

,weight,feed
1,179,horsebean
2,160,horsebean
3,136,horsebean
4,227,horsebean
5,217,horsebean


In [6]:
df.shape

(71, 2)

In [7]:
round(df.describe(),2)

,weight
count,71.00
mean,261.31
std,78.07
min,108.00
25%,204.50
50%,258.00
75%,323.50
max,423.00


The mean (261.31 lb) hovers just above the 50% mark (258.00 lb), so the mean is unlikely to be skewed by a few outliers. That said, the difference between the minimum and maximum is surprisingly high, indicating that the different types of feed have highly variable effects on chick weight. This statement is supported by how large the standard deviation is (78.07 lb), which is over 75% of the smallest chick weight (108.00 lb), meaning the data varies by almost an entire chicken for each sigma.

I would like to credit my colleague Stacy Keown for posting a solution to the index column issue that produced an inaccurate number of columns.

Github Repo Url: https://github.com/journeyrahman/cs82a-portfolio